### ISSUE TITLE
ss_feedback defaults to 'mean', which cannot load whenever scheduled sampling is on — four roster models hit it independently

### ISSUE THREAD

**Issue by Polichinel** [views-platform/views-hydranet#295]
`config_initializer.py:178` declares:

```python
ss_feedback: str = Field(default="mean")
```

Whenever scheduled sampling is active and `rollout_feedback` resolves to `'sample'`, that default is **guaranteed** to trip the C-259 guard at `config_initializer.py:1063`. A config that simply does not mention `ss_feedback` cannot load.

**Four views-models roster models hit this independently** — `blazing_meteor`, `blue_stranger`, `bright_starship`, `pink_pirate`. None could be loaded or evaluated at all. Each set `ss_schedule: 'linear'` and `ss_epsilon_max: 0.5` deliberately and simply never mentioned the third knob.

Fixed on the config side in **views-models#404** (merged as views-models#408) by declaring `'ss_feedback': 'sample'` in each. That is a symptom fix; this is the trap.

## Why it is the default and not the authors

The two guards already know the correct value. C-259 says so in its own error text:

> Set `ss_feedback` to match the resolved `rollout_feedback` (`'sample'`) — for a gated family head both are `'sample'`.

And the second guard at `:1078` rejects `'mean'` independently for any non-`self_zeroed` composition, because the training feedback would be ungated while inference's mean is gated.

So the system can *derive* the only acceptable value, states it in the failure message, and still defaults to a different one. Four configs fell in the same hole; that indicts the hole.

## What the roster actually looks like

Verified across every roster model:

| models | scheduled sampling | loaded before views-models#408? |
|---|---|---|
| `blazing_meteor`, `blue_stranger`, `bright_starship`, `pink_pirate` | **on** (`ss_epsilon_max: 0.5`) | ✗ |
| `violet_visitor`, `purple_alien`, `truncated_smoke` | **off** | ✓ |

**Every model that enabled scheduled sampling failed to load; every model that loaded had it off.** Until views-models#408, no model on the roster was running scheduled sampling under a config this repo would accept.

Worth stating plainly for #287/#288: whatever M30–M33 measured, the roster's SS-enabled configs were unloadable in their committed form on 2026-08-21.

## Options

1. **Default to the resolved `rollout_feedback`** rather than a literal. Makes the invalid state unrepresentable and matches what both guards already compute.
2. **No default — required whenever `ss_epsilon_max > 0`.** Louder, and forces the author to state the intent. Costs a line in every SS config.
3. **Leave it and rely on the guard.** Defensible only if the failure is cheap; four independent occurrences suggest it is not.

(1) is the smaller change and removes the class. (2) is more explicit but adds ceremony to configs that already carry two SS knobs.

## Timing — deliberately not proposed as an immediate change

`reports/2026-08-17_ss_retention_dossier` is a live pre-registered sweep (amendment 1 dated 2026-08-21) on whether correctly-configured scheduled sampling changes rollout retention, and **#288** is investigating whether the feedback path is differentiable at all. Changing a default that determines what SS *does* while both are in flight would be the wrong move.

Filed now so the finding is on the record next to #288, not to jump the queue.

## Interaction with #288, which may change the answer

#288's point 3 asks whether the gradient exists at all, given `_family_feedback_log1p` and a sampling step: *"A sample from a Gamma/NB draw is not differentiable without a reparameterisation."*

If `'sample'` feedback turns out to be non-differentiable — or if the `.detach()` at `training_engine.py:363,365` makes the question moot — then "default to the resolved `rollout_feedback`" is defaulting to something whose semantics are themselves under review. **This issue should not be actioned before #288 reports.** It is the config-surface half of the same subsystem, and the two answers should be consistent.

## One thing views-models has already done, so it is not repeated

The four configs now declare `'sample'` explicitly with the rule in a comment. That was verified against the real `HydraNetConfig` (all four load) and shown to be load-bearing by deleting the line and observing the exact C-259 error return. **The artifacts and published scores are unaffected** — `ss_feedback` governs training, not frozen weights — **but they can no longer be regenerated from those configs**, because they were trained under the effective `'mean'` and neither available fix reproduces that. Recorded in views-models#404.

Cross-refs: views-models#404, views-models#408, #287, #288.

**Comment by Polichinel** [comment]
## Labelled `pre-release` — but as the weakest item on that list, and here is the argument against

From the PyPI review (#340). Flagging because **a default is part of a published package's contract, and changing one after release is a behaviour change** — so this is the cheapest moment it will ever be fixable.

### Verified current state

`config_initializer.py:191` — `ss_feedback: str = Field(default="mean")`, while `rollout_feedback` defaults to `None` = AUTO, which resolves to `'sample'` for a registered family head. So the two defaults **contradict each other** for exactly the configuration this repo ships: a family head with scheduled sampling on.

### The argument against labelling it

**It fails loud, and the message is genuinely good** (`config_initializer.py:1219`):

> scheduled sampling is active (ss_schedule=…, ss_epsilon_max=…) but ss_feedback='mean' != resolved rollout_feedback='sample': training would feed back a different object than inference rolls out on (C-259). Set ss_feedback to match the resolved rollout_feedback…

A second validator (`:1234`) catches the gated-composition variant. Nobody is silently harmed: you get an error naming the fix. **This is an ergonomics defect, not a correctness one**, and on those grounds it could reasonably wait.

### The argument for

The trap already caught **four roster models independently**, all written by people with full context. An external user has less. And the fix — making the default `None` = AUTO so it resolves alongside `rollout_feedback`, rather than a literal that contradicts it — is the kind of change that is free before 0.1.0 and a semver event after.

### Recommendation

Fix it if the version-policy decision in #341 lands as "next release is 1.0.0", since the defaults should be right at the point they become a contract. **If 0.x continues, this can slip** — 0.x promises nothing, and the loud failure means the cost of slipping is an annoyed user rather than a wrong forecast.

Cross-refs: #340 · #341 (version policy — this depends on that decision) · risk register C-259, C-300.

**Comment by Polichinel** [comment]
`pre-release` label removed: 0.1.0 shipped on 2026-09-15 without this — no roster model uses scheduled sampling, so the default cannot bite in production. Still a real defect for the next SS experiment.
